base/boot.jl

```diff
@@ -269,9 +269,9 @@ Void() = nothing
 
 immutable VecElement{T}
     value::T
-    VecElement{T}(value::T) where T = new(value) # disable converting constructor in Core
+    VecElement{T}(value::T) where {T} = new(value) # disable converting constructor in Core
 end
-VecElement(arg::T) where T = VecElement{T}(arg)
+VecElement(arg::T) where {T} = VecElement{T}(arg)
```

Comment:
Presumably `Complex{Float64}(x::Float64) = ...` means the same as `Complex{T}(x::T) = ...` the `T` (whatever it is called) is a local variable that overrides any definition of `T` in the enclosing scope.

I don't recall people complaining about either of these things before.   `where` is great when you need triangular dispatch etc, but the old syntax still seems nice to me as an abbreviation.